### PR TITLE
feat(ripple): Implement subset of improved interaction response guidelines

### LINKED ISSUE #190: Implement subset of improved interaction response guidelines for ripple
-  _Tap_ - This could be a quick mousedown/mouseup, quick keydown/keyup, quick pointerstart/pointerend, trackpad tap, etc.
- _Tap and hold_ - Mousedown/keydown/pointerstart + hold + mouseup/keyup/pointerend

The rest of the improved interaction response guidelines will be covered in separate stories.

### PR DESCRIPTION
**Pull request by traviskaufman**
Background
===

When the ripple was originally designed from a motion perspective,
bounded and unbounded activation/deactivation used two different
animation styles, distinct from one another.

The UX updates to the ripple remove this distinction of bounded vs.
unbounded solely with regard to how the ripple animates in and out.
There is still a distinction between bounded vs. unbounded, but we no
longer have to account for this when animating the ripple. Thus, there
are no API changes; only UX changes.

Changes
===

- Implement UX changes for "tap" + "tap and hold" interactions
- Remove all references of "bounded" vs. "unbounded" from
  activation/deactivation code

Resolves #190

### CODE REVIEW

**Comment by traviskaufman** [comment]
@cristobalchao if you could do a quick pass on the UX here I'd appreciate it!

File: packages/mdc-ripple/_keyframes.scss [hunk 1]
```diff
@@ -20,46 +20,33 @@
 
 @keyframes mdc-ripple-fg-radius-in {
   from {
-    transform: translate(0) scale(1);
```
Comment (BBosman) [inline packages/mdc-ripple/_keyframes.scss]: Why is this removed? Don't we need a fallback for browsers that don't support var?

File: packages/mdc-ripple/_keyframes.scss [hunk 2]
```diff
@@ -20,46 +20,33 @@
...
     transform: translate(var(--mdc-ripple-fg-translate-start, 0)) scale(1);
     animation-timing-function: $mdc-animation-fast-out-slow-in-timing-function;
   }
 
   to {
```
Comment (BBosman) [inline packages/mdc-ripple/_keyframes.scss]: Same here.

File: packages/mdc-ripple/_mixins.scss [hunk 3]
```diff
@@ -215,29 +205,13 @@
...
-  &.mdc-ripple-upgraded--unbounded.mdc-ripple-upgraded--foreground-unbounded-deactivation#{$pseudo} {
-    animation:
-      mdc-ripple-fg-unbounded-opacity-deactivate 0 linear,
-      mdc-ripple-fg-unbounded-transform-deactivate 0 $mdc-animation-fast-out-slow-in-timing-function;
-    animation:
-      mdc-ripple-fg-unbounded-opacity-deactivate var(--mdc-ripple-fg-unbounded-opacity-duration, 0) linear,
-      mdc-ripple-fg-unbounded-transform-deactivate var(--mdc-ripple-fg-unbounded-transform-duration, 0) $mdc-animation-fast-out-slow-in-timing-function;
+  &.mdc-ripple-upgraded--foreground-deactivation#{$pseudo} {
+    // Retain transform from mdc-ripple-fg-radius-in activation
+    transform: translate(var(--mdc-ripple-fg-translate-end, 0)) scale(var(--mdc-ripple-fg-scale, 1));
```
Comment (BBosman) [inline packages/mdc-ripple/_mixins.scss]: Same here.

File: packages/mdc-ripple/_keyframes.scss (hunk 1 above)
Comment (BBosman) [inline packages/mdc-ripple/_keyframes.scss]: See #361.

Comment (traviskaufman) [inline packages/mdc-ripple/_keyframes.scss]: @BBosman good question. Typically, yes. However, this keyframe is only attached when `mdc-ripple-upgraded` is added to the ripple surface. Before we add that class, we check - within JS - to make sure custom properties are supported. Thus, fallbacks here aren't necessary.

...I'm going to add this explanation as a comment 😃

File: packages/mdc-ripple/_keyframes.scss (hunk 2 above)
Comment (traviskaufman) [inline packages/mdc-ripple/_keyframes.scss]: [ack](https://github.com/material-components/material-components-web/pull/419#discussion_r107239584)

File: packages/mdc-ripple/_mixins.scss (hunk 3 above)
Comment (traviskaufman) [inline packages/mdc-ripple/_mixins.scss]: [ack](https://github.com/material-components/material-components-web/pull/419#discussion_r107239584)

File: packages/mdc-ripple/_keyframes.scss (hunk 1 above)
Comment (BBosman) [inline packages/mdc-ripple/_keyframes.scss]: 👍

File: packages/mdc-ripple/constants.js [hunk 4]
```diff
@@ -15,42 +15,33 @@
  */
 
 export const ROOT = 'mdc-ripple';
-export const UPGRADED = `${ROOT}-upgraded`;
```
Comment (yeelan0319) [inline packages/mdc-ripple/constants.js]: nit: It seems like the exported const `ROOT` and `UPGRADED` is not being used anywhere.

**Comment by traviskaufman** [comment]
@yeelan0319 Changes made PTAL!

File: packages/mdc-ripple/_keyframes.scss [hunk 5]
```diff
@@ -20,46 +20,37 @@
...
+    // that custom properties are supported within the browser before adding this class, we can
+    // safely use them without a fallback.
     transform: translate(var(--mdc-ripple-fg-translate-start, 0)) scale(1);
     animation-timing-function: $mdc-animation-fast-out-slow-in-timing-function;
   }
 
   to {
-    transform: translate(0) scale(0);
-    transform: translate(var(--mdc-ripple-fg-translate-end, 0)) scale(var(--mdc-ripple-fg-scale, 0));
+    transform: translate(var(--mdc-ripple-fg-translate-end, 0)) scale(var(--mdc-ripple-fg-scale, 1));
```
Comment (yeelan0319) [inline packages/mdc-ripple/_keyframes.scss]: Is there any reason why the default value is different from `--mdc-ripple-fg-scale` default value (which is 0 and default to 0 else where in `_mixins.scss`). Is this a spec change?

File: packages/mdc-ripple/_mixins.scss (hunk 3 above)
Comment (yeelan0319) [inline packages/mdc-ripple/_mixins.scss]: Same thing here.

**Comment by yeelan0319** [comment]
@traviskaufman 
Brought this up in our stand up today as a parking lot since it seems like there is no further information about why `bounded` & `unbounded` is removed and I feel I am lacking of quite a few context information to give useful feedbacks on those implementation.

Moreover since so many changes have been made, I also want to make sure there is no changing to our API and we don't need to make any updates to `README.md`. 


Given we might need to review code across different components and git doesn't always clever enough to show diff correctly, I feel it might be helpful to provide slightly more high level description about the commit, I found a [guideline](https://github.com/blog/1943-how-to-write-the-perfect-pull-request) on Github that we might found useful.

**Comment by traviskaufman** [comment]
@yeelan0319 

When the ripple was originally designed from a motion perspective, `bounded` and `unbounded` activation/deactivation used two different animation styles, distinct from one another. 

The UX updates to the ripple _remove_ this distinction of `bounded` vs. `unbounded` _solely with regard to how the ripple animates in and out_. There is still a distinction between bounded vs. unbounded, but we no longer have to account for this when animating the ripple. Thus, there are no API changes; only UX changes. 

I'll go ahead and add this description above to the commit message.

File: packages/mdc-ripple/_keyframes.scss (hunk 5 above)
Comment (traviskaufman) [inline packages/mdc-ripple/_keyframes.scss]: Looks like an oversight on my part for the `--mdc-ripple-fg-scale` default value, which should be set to `1`. Will change.

File: packages/mdc-ripple/_mixins.scss (hunk 3 above)
Comment (traviskaufman) [inline packages/mdc-ripple/_mixins.scss]: [ack](https://github.com/material-components/material-components-web/pull/419/files#r108443126)

**Comment by traviskaufman** [comment]
@yeelan0319 changes made PTAL

**Comment by yeelan0319** [comment]
Cool. Thanks Travis. This feedback is useful 👍 🌟
